lib/exercism/user.rb

```diff
@@ -21,6 +21,12 @@ class User < ActiveRecord::Base
 
   before_save do
     self.key ||= Exercism.uuid
+
+    # migrate acls when track_mentor changes
+    if track_mentor_changed?
```

Comment:
I'm extremely aversive to AR callbacks in general. I find that making all sorts of extra things happen every time you save an object means that this happens in all the possible contexts that you haven't imagined. Tests. Batch scripts. Stuff like that. (How much do you want to bet that some company has sent a bajillion emails to their users by accident because they had a callback on some hook and then ran a batch script to migrate some data?)

Then these things get really complex and start causing callbacks to fail for strange and hard-to-debug reasons, and you get objects that aren't being persisted and don't even know where to begin looking for the failure.